src/interchange/src/protobuf/decode.rs

```diff
@@ -214,69 +217,121 @@ fn json_nested_from_serde_value(
     packer: &mut Row,
     f: &FieldDescriptor,
     descriptors: &Descriptors,
+    column_type: &ColumnType,
 ) -> Result<()> {
     packer.push(match val {
         SerdeValue::Bool(true) => Datum::True,
         SerdeValue::Bool(false) => Datum::False,
-        SerdeValue::I8(i) => json_number(i)?,
-        SerdeValue::I16(i) => json_number(i)?,
-        SerdeValue::I32(i) => json_number(i)?,
-        SerdeValue::I64(i) => json_number(i)?,
-        SerdeValue::U8(i) => json_number(i)?,
-        SerdeValue::U16(i) => json_number(i)?,
-        SerdeValue::U32(i) => json_number(i)?,
-        SerdeValue::U64(i) => json_number(i)?,
-        SerdeValue::F32(f) => json_number(f)?,
-        SerdeValue::F64(f) => json_number(f)?,
+        SerdeValue::I8(i) => Datum::Int32(*i as i32),
```

Comment:
The problem here is that we don't have internal types that would map to those Protobuf types, right? Was there any discussion on what we should use to represent those types that we can't natively support? (Sorry if I missed that. :sweat_smile: )

Reply:
Hm. So I actually just copied over [what we do for non-nested numeric types already](https://github.com/MaterializeInc/materialize/blob/main/src/interchange/src/protobuf/decode.rs#L131). My read of `json_number` was that we were converting the nested values to be `jsonb` friendly, but I could definitely be wrong!

@quodlibetor it looks like you are partially responsible for this code, do you have thoughts?

Reply:
oh, and @ruchirK !

Reply:
Okay, so it took me all day, but I finally got what you meant 😂. I've fixed this! 

(I deleted my other comments hoping that it won't ping the other folks on Monday!)